ff/serving/coordinator/coordinator.go

```diff
@@ -254,56 +258,46 @@ func (c *Coordinator) runRegisterSourceJob(resID metadata.ResourceID) error {
 
 //should only be triggered when we are registering an ONLINE feature, not an offline one
 func (c *Coordinator) runFeatureMaterializeJob(resID metadata.ResourceID) error {
+	c.Logger.Debug("Running feature materialization job on resource: ", resID)
 	feature, err := c.Metadata.GetFeatureVariant(context.Background(), metadata.NameVariant{resID.Name, resID.Variant})
 	if err != nil {
 		return err
 	}
 	status := feature.Status()
 	featureType := feature.Type()
 	if status == metadata.READY || status == metadata.FAILED {
-		return fmt.Errorf("feature already set to %s", metadata.ResourceStatus(status))
+		return fmt.Errorf("feature already set to %s", status.String())
 	}
-	if err := c.Metadata.SetStatus(context.Background(), resID, string(metadata.PENDING)); err != nil {
+	if err := c.Metadata.SetStatus(context.Background(), resID, pb.ResourceStatus{Status: pb.ResourceStatus_PENDING}); err != nil {
 		return err
 	}
 	sourceNameVariant := feature.Source()
-	source, err := c.Metadata.GetSourceVariant(context.Background(), sourceNameVariant)
+	source, err := c.AwaitPendingSource(sourceNameVariant, 30) //second argument is seconds to wait for source to become ready
```

Comment:
Could add this timeout value to the coordinator config so it can be changed further up

Reply:
sure, I was just unsure where to put it as there are multiple timeout levels that may not be obvious from a top down perspective what each one corresponds to. Though a global timeout value with optional overrides would make sense

Reply:
Yeah that would work